src/ContinuosBondingERC20Token.sol

```diff
@@ -201,45 +208,54 @@ contract ContinuosBondingERC20Token is ERC20, ReentrancyGuard {
-        if (poolType == LP_POOL.Uniswap) {
-            wNative = IUniswapV2Router02(router).WETH();
-
-            address pair = factory.getPair(address(this), wNative);
-            if (pair == address(0)) pair = factory.createPair(address(this), wNative);
-            IWETH(wNative).deposit{ value: currentEth }();
-            IWETH(wNative).transfer(pair, currentEth);
-            _transfer(address(this), pair, currentTokenBalance);
-            liquidity = IPair(pair).mint(address(this));
-        } else if (poolType == LP_POOL.TraderJoe) {
-            wNative = IJoeRouter02(router).WAVAX();
-
-            address pair = factory.getPair(address(this), wNative);
-            if (pair == address(0)) pair = factory.createPair(address(this), wNative);
-            IWETH(wNative).deposit{ value: currentEth }();
-            IWETH(wNative).transfer(pair, currentEth);
-            _transfer(address(this), pair, currentTokenBalance);
-            liquidity = IPair(pair).mint(address(this));
+        // Create the pool if it doesn't exist
+        (address token0, address token1) = address(this) < WETH ? (address(this), WETH) : (WETH, address(this));
+        address pool = nonfungiblePositionManager.createAndInitializePoolIfNecessary(token0, token1, 30000, _getSqrtPriceX96(currentEth, currentTokenBalance));
+
+        // Approve the position manager to spend tokens
+        _approve(address(this), address(nonfungiblePositionManager), currentTokenBalance);
+        IWETH(WETH).deposit{value: currentEth}();
+        IWETH(WETH).approve(address(nonfungiblePositionManager), currentEth);
+
+        // Add liquidity
+        INonfungiblePositionManager.MintParams memory params = INonfungiblePositionManager.MintParams({
+            token0: address(this),
+            token1: WETH,
+            fee: 3000,
+            tickLower: TickMath.MIN_TICK,
+            tickUpper: TickMath.MAX_TICK,
+            amount0Desired: currentTokenBalance,
+            amount1Desired: currentEth,
+            amount0Min: 0,
+            amount1Min: 0,
+            recipient: address(this),
+            deadline: block.timestamp
+        });
+
+        (uint256 tokenId, uint128 liquidity, uint256 amount0, uint256 amount1) = nonfungiblePositionManager.mint(params);
+
+        // Transfer the NFT to the treasury if feeRecipient is not zero
+        if (IBondingERC20TokenFactory(factory).feeRecipient() != address(0)) {
+            nonfungiblePositionManager.transferFrom(address(this), TREASURY_ADDRESS, tokenId);
         }
-        // Burn the LP tokens received
-        IERC20 lpToken = IERC20(factory.getPair(address(this), wNative));
-        lpToken.transfer(BURN_ADDRESS, liquidity);
 
-        emit PairCreated(currentEth, currentTokenBalance, liquidity, address(lpToken));
+        emit PairCreated(amount1, amount0, liquidity, pool);
+    }
+
+    function _getSqrtPriceX96(uint256 amountToken, uint256 amountWETH) internal pure returns (uint160) {
```

Comment:
rename the function arguments to `token0` and `token1` and refactor this function to calculate price accordingly.